python/common/org/python/types/List.java

```diff
@@ -337,14 +337,22 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                     if (step < 0) {
                         long start;
                         if (slice.start != null) {
-                            start = Math.min(slice.start.value, this.value.size());
+                            if (slice.start.value < 0) {
+                                start = this.value.size() + slice.start.value;
```

Comment:
@uranusjr This line will only be executed for the case where `start` is less than 0. But you're right that it still needs the Math.max around it, for the case when start is lesser than `-1 -
 size of the list`. =)

@Nasreen123 thanks for contributing! :tada: 
Can you please add a test for the case `x[- 5 - len(x): -1]` ?